binder/postBuild

```diff
@@ -0,0 +1,2 @@
+python -m pip install --upgrade -r requirements-dev.txt
+python -m pip install .
```

Comment:
I tried the binder button in your PR message and it takes a long time to build iminuit. I think it should not build iminuit at all, just install a wheel from a release (`python -m pip install iminuit` should do it). Also, we can cut down the number of installed dependencies, not everything in requirements-dev.txt is needed for each notebook. Is there a way to specify the dependencies explicitly for each notebook?

Reply:
I am aware that `python -m pip install iminuit` always installs the latest release, so we give up on syncing the notebooks with the current development version. But maybe that is acceptable.

Reply:
Any changes to iminuit should be backward compatible or have a gracious deprecation period.

Reply:
Unless there is a cool binder way of doing it, I think we should add an installation variant so that you can do `python -m pip install iminuit[tutorial]` to install the dependencies for running the tutorials.

Reply:
The idea here is that with

```
python -m pip install .
```

it is installing directly from the `develop` branch which, while slower, only needs to be done once after a PR. I'm all for adding an additional 'tutorial' extra though beyond `tests`. That seems beyond the scope of this PR, so I can open another one to add this.

Reply:
What is the intended audience for binder notebooks, then? I thought this is for people who are curious about trying out iminuit, but don't want to install it. They can click on the binder button and try out the tutorials. If that was me, and running binder takes forever to install iminuit, I would be put off already. That's why I think the binder books must run fast.

I don't need the binder for PRs.

Reply:
> I'm all for adding an additional 'tutorial' extra though beyond tests. That seems beyond the scope of this PR, so I can open another one to add this.

No need, I am currently working on that in another PR.

Reply:
> They can click on the binder button and try out the tutorials. If that was me, and running binder takes forever to install iminuit, I would be put off already. That's why I think the binder books must run fast.

The pyhf Binders run fast because they are rebuilt with each PR. Binder is _always_ going to be slow to build, but the images are cached on whatever Binder federation cluster got your request once they are, so once they are built then they will load as fast as possible.

> I thought this is for people who are curious about trying out iminuit, but don't want to install it.

Yes, we're thinking of the same audience. I'm just expecting and whoever authors a PR either uses CI to autotrigger builds after PRs are merged or that they remember to click the badge after their PR goes in so that the target audience has the minimal wait time possible, with the current features of `iminuit`.

Though with pyhf we are still working on adding in functionality, so we sometimes want to demo notebooks and things before a new release. With iminuit it seems that you have a more mature library, so perhaps there really isn't the need to have it install from `develop`'s head and doing
```
python -m pip install iminuit[tutorial]
```
is the desired way forward regardless of build time as they the user is seeing truly what is available to them _right now_ in the release.

Reply:
> Yes, we're thinking of the same audience. I'm just expecting and whoever authors a PR either uses CI to autotrigger Binder builds after PRs are merged

Ok then, how would I add that? I don't want to remember to click on a button.

Reply:
Anyway, automatically triggering the Binder build is another PR so I am going to merge this one.